vector/src/main/java/im/vector/app/features/home/room/list/home/HomeRoomListViewModel.kt

```diff
@@ -78,16 +80,45 @@ class HomeRoomListViewModel @AssistedInject constructor(
     private val _sections = MutableSharedFlow<Set<HomeRoomSection>>(replay = 1)
     val sections = _sections.asSharedFlow()
 
+    private val filtersPreferencesFlow = MutableSharedFlow<Boolean>(replay = 1)
+
     private var filteredPagedRoomSummariesLive: UpdatableLivePageResult? = null
 
     init {
         configureSections()
+        observePreferences()
+    }
+
+    private fun observePreferences() {
+        preferences.registerFiltersListener { areFiltersEnabled ->
+            viewModelScope.launch {
+                filtersPreferencesFlow.emit(areFiltersEnabled)
```

Comment:
Is this triggering immediately right after you register the listener? If so, we don't need the below lines.

Reply:
For some reason it doesn't. If I remove that emit, it will not show filters until you change setting

Reply:
-> `DataStore` :)